File: collected_code/auto/aiml_ds/AIML_002/python_standard/run_1/code/routes/predict.py
from flask import Blueprint, request, jsonify
from models.model_loader import ModelLoader

bp = Blueprint('predict', __name__)
model_loader = ModelLoader()

@bp.route('/predict', methods=['POST'])
def predict():
    data = request.get_json()
    if not data or 'features' not in data:
        return jsonify({'error': 'Missing features in request'}), 400
    
    features = data['features']
    if not isinstance(features, list):
        return jsonify({'error': 'Features must be a list'}), 400
    
    try:
        model = model_loader.get_model()
        prediction = model.predict([features])[0]
        return jsonify({'prediction': float(prediction)})
    except Exception as e:
        return jsonify({'error': str(e)}), 500

File: collected_code/auto/aiml_ds/AIML_002/typescript_standard/run_1/code/src/routes/predict.ts
import { Router, Request, Response } from 'express';
import { ModelLoader } from '../models/ModelLoader';

const router = Router();
const modelLoader = new ModelLoader();

router.post('/', async (req: Request, res: Response) => {
  try {
    const { features } = req.body;
    
    if (!features || !Array.isArray(features)) {
      return res.status(400).json({ error: 'Invalid input: features must be an array' });
    }

    const model = await modelLoader.loadModel();
    const prediction = await modelLoader.predict(model, features);
    
    res.json({ prediction });
  } catch (error) {
    res.status(500).json({ error: 'Prediction failed', message: error instanceof Error ? error.message : 'Unknown error' });
  }
});

export default router;
